### PR TITLE
Skip pre-checks in favor of pre-commit CI.

### PR DESCRIPTION
**Pull request by bdice**
## Description
I recently enabled pre-commit CI. This can replace the pre-checks currently being run on CircleCI.

## Motivation and Context
Removes redundancy and potentially speeds up CI.

## Types of Changes
- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

### CODE REVIEW

**Comment by tommy-waltmann** [comment]
Does the other PR use github actions to do pre-commit CI? If CircleCI is handling the pre-commit, we may want to require the pre-commit checks pass before running tests to conserve CI time.

**Comment by bdice** [comment]
> 
> 
> Does the other PR use github actions to do pre-commit CI? If CircleCI is handling the pre-commit, we may want to require the pre-commit checks pass before running tests to conserve CI time.

Pre-commit CI was added as an application, so there is no corresponding PR. You can see it here (may require some kind of permissions, I'm not sure): https://github.com/glotzerlab/signac/settings/installations

The Pre-commit CI service is better than running the pre-commit hooks on CircleCI because it can also update the PR to fix any formatting issues and also keeps the pre-commit configuration updated.

Your comment about conserving CI time is possible, but I think that on average we will see a reduction in time spent waiting for CI if we remove the first step (style checks) that was a dependency for all other steps. The time I would measure is "time from pushing a commit to all checks passing." Concurrency limits on CircleCI (4 runners for the glotzerlab organization) may be a limiting factor some of the time, e.g. if pushes occur to signac, signac-flow, and freud at the same time, but I believe the majority of commits to the repository do pass the style checks so parallelizing that step is helpful. Also, if pre-commit CI catches any style problems, it will push to the branch, which will cancel any queued tests on CircleCI for the "unstylish" commit and queue up tests for the fixed commit.

File: .circleci/config.yml [hunk 1]
```diff
@@ -268,31 +268,15 @@ workflows:
...
-      - linux-pypy-37:
-          requires:
-            - pre-checks
-      - windows-python-38:
-          requires:
-            - pre-checks
-      - macos-python-3:
-          requires:
-            - pre-checks
+#     - pre-checks
```
Comment (csadorf) [inline .circleci/config.yml:271]: Why do we keep this comment?

**Review by csadorf — APPROVED** [review]
One question, but LGTM overall.

**Comment by joaander** [comment]
> Your comment about conserving CI time is possible, but I think that on average we will see a reduction in time spent waiting for CI if we remove the first step (style checks) that was a dependency for all other steps. The time I would measure is "time from pushing a commit to all checks passing." Concurrency limits on CircleCI (4 runners for the glotzerlab organization) may be a limiting factor some of the time, e.g. if pushes occur to signac, signac-flow, and freud at the same time, but I believe the majority of commits to the repository do pass the style checks so parallelizing that step is helpful. Also, if pre-commit CI catches any style problems, it will push to the branch, which will cancel any queued tests on CircleCI for the "unstylish" commit and queue up tests for the fixed commit.

CircleCI still limits 4 running jobs? If this is every a limit, we should switch to GitHub Actions. Concurrency on GitHub Actions is 60 (though only 5 on mac) for our free Team account: https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits

**Review by tommy-waltmann — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (bdice) [reply .circleci/config.yml:271]: I was planning to keep this temporarily -- I don't want to remove the `pre-checks` job configuration just yet, until we keep pre-commit CI enabled for a little bit longer.

**Pull request merged**